src/onig-reg-exp.cc

```diff
@@ -10,6 +10,17 @@ OnigRegExp::OnigRegExp(const string& source)
   lastSearchPosition = -1;
   lastSearchResult = NULL;
 
+  hasGAnchor = false;
+  for (size_t pos = 0, len = source.size(); pos < len; pos++) {
+    if (source[pos] == '\\' && pos + 1 < len) {
+      if (source[pos + 1] == 'G') {
+        hasGAnchor = true;
+        break;
+      }
+      pos++;
+    }
+  }
+
```

Comment:
I think this could be written using [`std::string::find`](http://www.cplusplus.com/reference/string/string/find/), which should be faster for long regular expressions:

```cpp
hasGAnchor = source.find("\\G") != string::npos;
```

Reply:
:+1: for the speed benefit, but would that not fall for an escaped backslash followed by G?

e.g. `\\Ghello` vs `\Ghello`. The first is literally looking for `\Ghello`, while the second is looking for `hello` if the search position is right before `h`.

Reply:
Ah, of course; I didn't notice that you were actually accounting for that case here.